src/ui/src/metadata_widget.cpp

```diff
@@ -1,10 +1,35 @@
 #include "metadata_widget.h"
 
-#include "ui_metadata_widget.h"
-
 MetadataWidget::MetadataWidget(QWidget* parent)
-		: QDockWidget(parent), ui_(new Ui::MetadataWidget) {
-	ui_->setupUi(this);
+		: QTableWidget(parent) {
+	
+	setColumnCount(2);
+	setHorizontalHeaderLabels(lables_);
+	// Test();
+}
+
+void MetadataWidget::OnShowMetadata(const MetadataPack& pack) {
+	setRowCount(pack.size());
```

Comment:
Narrowing conversion from `size_t` (unsigned long long) to `int`. Escape such conversions or make them explicit